server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -40,6 +42,7 @@ public class EsExecutors {
 
     // although the available processors may technically change, for node sizing we use the number available at launch
     private static final int MAX_NUM_PROCESSORS = Runtime.getRuntime().availableProcessors();
+    private static final Pattern THREAD_NAME_PATTERN = Pattern.compile("elasticsearch(\\[([^]]+)])?\\[([^]]+)]\\[T#\\d+]");
```

Comment:
I know some people have a visceral reaction to regexes and beyond a certain complexity I agree, but this one seems simple enough.